src/components/Checkbox/Checkbox.tsx

```diff
@@ -42,23 +44,28 @@ export const Checkbox = ({
       )}
       htmlFor={inputId}
     >
-      {!readOnly && (
-        <span className={classes['checkbox-wrapper']}>
-          <input
-            checked={checked ?? false}
-            className={classes.checkbox}
-            disabled={disabled}
-            id={inputId}
-            name={name}
-            onChange={disabled ? undefined : onChange}
-            type='checkbox'
-          />
-          <span className={classes['visible-box']}>
-            <span className={classes['visible-box__checkmark']} />
+      <span className={classes['checkbox-and-label']}>
+        {!readOnly && (
+          <span className={classes['checkbox-wrapper']}>
+            <input
+              checked={checked ?? false}
+              className={classes.checkbox}
+              disabled={disabled}
+              id={inputId}
+              name={name}
+              onChange={disabled ? undefined : onChange}
+              type='checkbox'
+            />
+            <span className={classes['visible-box']}>
+              <span className={classes['visible-box__checkmark']} />
+            </span>
           </span>
-        </span>
+        )}
+        {label && <span className={classes.label}>{label}</span>}
+      </span>
+      {description && (
+        <span className={classes['description']}>{description}</span>
```

Comment:
We should probably add an ID to this, and link the input to this field with [`aria-describedby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby#:~:text=The%20aria%2Ddescribedby%20attribute%20lists,the%20text%20that%20describes%20them.)? Not sure how this will work since its all wrapped within the `label` though... Perhaps we need to change the markup a bit here. WDYT?

Reply:
As I understand form reading the docs, `aria-describedby` shouldn't care about `<label>`. On the other hand, I think we should also add `aria-labelledby` to `<span className={classes.label}>`, to exclude the description text from the accessible name. According to the docs, it has precedence over `<label>` when calculating the accessible name, while not overriding `<label>`'s click functionality.